http-server-netty/src/test/groovy/io/micronaut/http/server/netty/binding/QueryParameterFormattingSpec.groovy

```diff
@@ -0,0 +1,183 @@
+package io.micronaut.http.server.netty.binding
+
+import io.micronaut.core.annotation.Introspected
+import io.micronaut.core.convert.format.Format
+import io.micronaut.http.HttpMethod
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.MediaType
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.QueryValue
+import io.micronaut.http.client.exceptions.HttpClientResponseException
+import io.micronaut.http.server.netty.AbstractMicronautSpec
+import org.reactivestreams.Publisher
+import reactor.core.publisher.Flux
+import spock.lang.Unroll
+
+public class QueryParameterFormattingSpec extends AbstractMicronautSpec {
+    private static String PIPE = "%7C";
+
+    @Unroll
+    void "test bind formatted query parameters for URI #uri"() {
+        given:
+        var encodedUri = uri.replace("|", PIPE).replace(" ", "+")
+        HttpRequest<?> req = HttpRequest.create(HttpMethod.GET, encodedUri)
+        Publisher<HttpResponse<?>> exchange = rxClient.exchange(req, String)
+        HttpResponse<?> response = Flux.from(exchange).blockFirst()
+        def body = response.body()
+
+        expect:
+        body == result
+
+        where:
+        uri                                                           | result
+        // List
+        '/formatted/csv?param=a,b,c'                                  | ["a", "b", "c"].inspect()
+        '/formatted/ssv?v=a b d'                                      | ["a", "b", "d"].inspect()
+        '/formatted/pipes?param=what|is|life'                         | ["what", "is", "life"].inspect()
+        '/formatted/multi?value=one&value=two&value=3'                | ["one", "two", "3"].inspect()
+        '/formatted/deep?param[0]=a&param[1]=b'                       | ["a", "b"].inspect()
+        '/formatted/csv'                                              | [].inspect()
+        '/formatted/csv?param='                                       | [].inspect()
```

Comment:
It would make more sense to me if the `?param=` was omitted in this case

Reply:
I did it to test how it works in corner cases. I don't think there should be any kind of exception if user does write it in this way.

Reply:
I agree there shouldn't be an exception. I was asserting the `?param=` should not be present in the URI